nearcache/types.go

```diff
@@ -0,0 +1,131 @@
+	EvictionPolicyLFU
+	// EvictionPolicyNone removes no entries.
+	EvictionPolicyNone
+	// EvictionPolicyRandom removes a random entry.
+	EvictionPolicyRandom
+	evictionPolicyCount // internal
+)
+
+// String returns a string representation of the eviction policy.
+func (p EvictionPolicy) String() string {
+	switch p {
+	case EvictionPolicyLRU:
+		return "LRU"
+	case EvictionPolicyLFU:
+		return "LFU"
+	case EvictionPolicyNone:
+		return "NONE"
+	case EvictionPolicyRandom:
+		return "RANDOM"
+	}
+	panic(fmt.Errorf("unknown eviction policy: %d", p))
+}
+
+// Stats contains statistics for a Near Cache instance.
+type Stats struct {
+	// Evictions is the number of evictions.
+	Evictions int64
+	// Evictions is the number of expirations.
+	Expirations int64
+	// Hits is the number of times a key was found in the Near Cache.
+	Hits int64
+	// InvalidationRequests is the number of invalidation requests.
+	// An invalidation request may be successful or not.
+	InvalidationRequests int64
+	// Invalidations is the number of successful invalidations.
+	Invalidations int64
+	// Misses is the number of times a key was not found in the Near Cache.
+	Misses int64
+	// OwnedEntryCount is the number of entries in the Near Cache.
+	OwnedEntryCount int64
+	// OwnedEntryMemoryCost is the estimated memory cost of the entries in the Near Cache.
+	OwnedEntryMemoryCost int64
+	// LastPersistenceKeyCount is the number of keys saved in the last persistence task.
+	LastPersistenceKeyCount int64
+	// LastPersistenceWrittenBytes is the size of the last persistence task.
+	LastPersistenceWrittenBytes int64
+	// PersistenceCount is the number of completed persistence tasks.
+	PersistenceCount int64
+	// LastPersistenceTime is the time of the last completed persistence task.
+	LastPersistenceTime time.Time
+	// LastPersistenceDuration is the duration of the last completed persistence task.
+	LastPersistenceDuration time.Duration
+	// LastPersistenceFailure is the error message of the last completed persistence task.
+	LastPersistenceFailure string
+	// CreationTime is the time the Near Cache was initialized.
+	CreationTime time.Time
+}
+
+// Ratio returns the ratio of hits to misses.
+// Returns math.Nan if there were no hits or misses.
+// Otherwise returns +math.Inf if there were no misses.
+func (s Stats) Ratio() float64 {
+	if s.Misses == 0 {
+		if s.Hits == 0 {
+			return math.NaN()
+		}
+		return math.Inf(1)
+	}
+	return (float64(s.Hits) / float64(s.Misses)) * 100.0
+}
+
+// EvictionPolicyComparator is used for comparing entries to be evicted.
+type EvictionPolicyComparator interface {
+	// Compare returns a negative integer if a is less than b, 0 if a is equal to b or a positive integer if a is greater than b.
+	Compare(a, b EvictableEntryView) int
+}
+
+// EvictableEntryView is the contract point from the end user perspective for serving/accessing entries that can be evicted.
+type EvictableEntryView struct {
```

Comment:
Commenting just to make static check visible

Reply:
I've converted this one to an interface.